app/main.py

```diff
@@ -1,12 +1,26 @@
+import hashlib
+
 from flask import abort, Flask, jsonify, request
 
-from app.conf import DEBUG
-from app.memorize import CacheMissError
+from app import conf
+from app.util import make_queue
+from app.memcached import memcached
 from app.search.recommendation import SearchRecommendation
 
 
 application = Flask(__name__)
-application.config['DEBUG'] = DEBUG
+application.config.update(
+    CELERY_BROKER_URL=conf.CELERY_BROKER_URL,
+    DEBUG=conf.DEBUG
+)
+queue = make_queue(application)
+
+
+@queue.task(name='main.recommend')
+def recommend(q, key):
+    recommendation = SearchRecommendation(q).do_search(q)
```

Comment:
I don't think so; we'll eventually want to use this to refresh cache in a [beat](http://docs.celeryproject.org/en/latest/userguide/periodic-tasks.html) process.